src/aiida/cmdline/commands/cmd_node.py

```diff
@@ -424,7 +424,7 @@ def verdi_graph():
 
 
 @verdi_graph.command('generate')
-@arguments.NODE('root_node')
+@arguments.NODES('root_nodes')
```

Comment:
What I said before about the `--` marker was correct but really applies to _options_ and the root nodes are _arguments_ so the approach won't work. Making the root node argument take multiple values is problematic and can never work. There is just no way for the parser to know whether the last value is a node identifier or the optional output filename.

We have two alternatives here:
1. Add a new _option_ to specify multiple root nodes (deprecating the old positional argument)
2. Add a new option to specify the output filename that replaces the optional positional argument

I think the first one is probably best, because we can relatively easily deprecate the old behavior so we don't break current functionality.

Essentially you add `@options.NODES()`, this will add the `nodes` argument to the `graph_generate` function definition. Then you check in the function definition if `root_node` is defined and if so, you raise a deprecation warning saying that nodes should now be specified with `--nodes/-N ...`. The deprecation can be done using:
```python
from aiida.cmdline.utils import echo

echo.echo_deprecated('Specifying the root node positionally is deprecated, please use the `-N/--nodes` options instead')
```


And then you do:
```python
root_nodes = nodes or [root_node]
```
(maybe we should first check if `nodes` and `root_node` are _both_ defined and issue a warning, because it means the user did something like `verdi node graph generate -N 10 11 -- 15`).

Reply:
I implemented this, thanks for your help.

One issue remains: when using the new `--nodes` option you can only set an output file name like this (specifying the positonal root node which in turn gets ignored):

```bash
$ verdi node graph generate -N 5673 5666 -- 123 foo.pdf
Warning: Specifying the root node positionally and the `-N/--nodes` option at the same time is not supported, ignoring the root node! Please use the `-N/--nodes` option only
Success: Output written to `foo.pdf`
```

Reply:
Thanks @bilke . I had a look at this. I think our best bet is to also deprecate the `OUTPUT_FILE` argument and replace it with an option. I added this in a last commit. So now a user wanting multiple root nodes can use `verdi node graph generate -N 1 2 -O custom_output.pdf`. I added some tests and also tested manually and it seems to work ok. Let me know what you think of this solution.

Reply:
That is perfectly fine for me!